routes/helpers/render.js

```diff
@@ -61,7 +61,15 @@ let render = {
     vars.qualifiedURL = config.qualifiedURL;
 
     vars.urlPath = url.parse(req.originalUrl).pathname;
```

Comment:
I would suggest a separate variable for the path that's appended to the login/register links. Moreover, I would recommend that if there is already a returnTo in the query string, to use it as the value for that variable, rather than setting it here. I haven't tested that approach, but I think it would work.

The reason is that right now, if you go to /signin and _then_ go to /register (or click either button twice), it will _still_ pass along the "sign in" or "register" form as a return-to parameter, which is confusing. We should never, ever be redirected to a sign-in or register form after signing in or registering.

Consider the Wikipedia behavior:

1) Click https://en.wikipedia.org/wiki/Main_Page
2) Click the "Log in" link
3) Note how "returnto=Main+Page" is appended to the URL
4) Click the "Log in" link again
5) Note how "returnto=Main+Page" is _still_ appended to the URL
6) Click the "Create account" link
7) Note how "returnto=Main+Page" is _still_ appended to the URL

That is the behavior we should aim to reproduce.

Reply:
This is what I aimed for but need to work out the cases for the if block.  I could check the path to not rewrite the returnTo on the signin and registration unless empty(from an external site).

The other way to handle the sign in/ registration page buttons would be to remove them.  They're redundant on these pages.